packages/client/src/runtime/core/engines/data-proxy/utils/request.ts

```diff
@@ -87,7 +87,16 @@ function buildOptions(options: RequestOptions): Https.RequestOptions {
 function buildResponse(incomingData: Buffer[], response: IncomingMessage): RequestResponse {
   return {
     text: () => Promise.resolve(Buffer.concat(incomingData).toString()),
-    json: () => Promise.resolve(JSON.parse(Buffer.concat(incomingData).toString())),
+    json: () => {
+      const dataAsString = Buffer.concat(incomingData).toString()
+      let dataAsJson
+      try {
+        dataAsJson = JSON.parse(dataAsString)
+        return Promise.resolve(dataAsJson)
+      } catch (e) {
+        return Promise.reject(`Failed to parse JSON: ${e.message} - data.length=${dataAsString.length}`)
```

Comment:
Why did you need to change this line of code?

Reply:
To understand what's going on with the unexpected "Unexpected end of JSON input" errors.
It turns out the data is empty when that happens.